src/main/java/com/netflix/imflibrary/st0429_8/PackingList.java

```diff
@@ -138,44 +120,121 @@ public PackingList(ResourceByteRangeProvider resourceByteRangeProvider, @Nonnull
+                streamSources[1] = new StreamSource(pkl_is);
+
+                SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+                Schema schema = schemaFactory.newSchema(streamSources);
+
+                ValidationEventHandlerImpl validationEventHandlerImpl = new ValidationEventHandlerImpl(true);
+                JAXBContext jaxbContext = JAXBContext.newInstance(supportedPKLSchemas.get(i).getPKLContext());
+                Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+                unmarshaller.setEventHandler(validationEventHandlerImpl);
+                unmarshaller.setSchema(schema);
+
+                packingListTypeJAXBElement = (JAXBElement) unmarshaller.unmarshal(inputStream);
+                pklSchema = supportedPKLSchemas.get(i);
+
+                if (validationEventHandlerImpl.hasErrors()) {
+                    List<ValidationEventHandlerImpl.ValidationErrorObject> errors = validationEventHandlerImpl.getErrors();
+                    for (ValidationEventHandlerImpl.ValidationErrorObject error : errors) {
+                        imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_PKL_ERROR, error.getValidationEventSeverity(), error.getErrorMessage());
+                    }
+                    throw new IMFException(validationEventHandlerImpl.toString());
+                }
+                break;
+            }
+            catch (SAXException | JAXBException e){
+                if(i == supportedPKLSchemas.size()-1){
+                    throw e;
                 }
-                throw new IMFException(validationEventHandlerImpl.toString());
             }
+        }
+        this.pklSchema = pklSchema;
+        this.packingListTypeJAXBElement = packingListTypeJAXBElement;
+
+        switch(this.pklSchema.getPKLContext()) {
+            case "org.smpte_ra.schemas.st0429_8_2007.PKL":
+                //this.packingListType = PackingList.checkConformance(packingListTypeJAXBElement.getValue());
+                org.smpte_ra.schemas.st0429_8_2007.PKL.PackingListType packingListType = (org.smpte_ra.schemas.st0429_8_2007.PKL.PackingListType) this.packingListTypeJAXBElement.getValue();
+                this.uuid = UUIDHelper.fromUUIDAsURNStringToUUID(packingListType.getId());
+
+                for (org.smpte_ra.schemas.st0429_8_2007.PKL.AssetType assetType : packingListType.getAssetList().getAsset()) {
+                    Asset asset = new Asset(assetType.getId(), Arrays.copyOf(assetType.getHash(), assetType.getHash().length), assetType.getSize().longValue(), assetType.getType(), assetType.getOriginalFileName().getValue());
+                    this.assetList.add(asset);
+                }
+                break;
+            case "org.smpte_ra.schemas.st0429_8_2016.PKL":
+                throw new IMFException(String.format("Please check the PKL document and namespace URI, currently we only support the 2007 PKL schema URI"));
```

Comment:
i might be having a memory lapse here - is there a reason why do not have an implementation for the 2016 case? Is it because we have not generated the model yet?